File: AIO/stacks/stacks.cpp
#include <fstream>
#include <vector>

using namespace std;
int main() {
	ifstream cin("stacksin.txt");
	ofstream cout("stacksout.txt");
	string n;
	vector<string> s;

	cin >> n;
	while (n != "q") {
		if (n == "p") {
			s.pop_back();

		}
		else {
			s.push_back(n);
		}
		if (s.size()) {
			for (string k : s) {
				cout << k << " ";
			}
			cout << "\n";
		}
		else {
			cout << "stack is empty!\n";
		}
		cin >> n;
	}
}

File: AIO/stacks/stacks.py
infile=open('stacksin.txt','r')
outfile=open('stacksout.txt','w')
list2=[]
string=''
for c in infile:
    i=c.strip()
    if i.isalpha()==False:
        list2.append(i)
        string+=str(' '.join(list2)+'\n')
    elif i=='p':
        list2.pop()
        if len(list2)==0:
            string+='stack is empty!\n'
        else:
            string+=str(' '.join(list2)+'\n')
    elif i=='q':
        break
outfile.write(string)
outfile.close()


